src/components/v5/common/ActionSidebar/partials/MultiSigSidebar/partials/MultiSigWidget/utils.ts

```diff
@@ -84,3 +84,31 @@ export const getAllUserSignatures = (
 
   return allUserSignatures;
 };
+
+export const getSignaturesPerRole = (
+  signatures: MultiSigUserSignature[],
+): {
+  approvalsPerRole: Record<number, MultiSigUserSignature[]>;
+  rejectionsPerRole: Record<number, MultiSigUserSignature[]>;
+} => {
+  const approvalsPerRole = {};
+  const rejectionsPerRole = {};
+
+  signatures.forEach((signature) => {
+    const { role, vote } = signature;
+
+    if (vote === 'Approve') {
+      if (!approvalsPerRole[role]) {
+        approvalsPerRole[role] = [];
+      }
+      approvalsPerRole[role].push(signature);
+    } else if (vote === 'Reject') {
```

Comment:
Can we have `enum`s for Approve / Reject ?

Reply:
we already have, I will fix it while rebasing it, nice catch